### PR TITLE
topology: byt: Add virtual widgets

### PR DESCRIPTION
**Pull request by ranj063**
Add the required virtual widgets to the BYT codec topology
to suppress the errors seen due to the dapm routes set up
in the machine driver.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>

### CODE REVIEW

**Comment by ranj063** [comment]
This replaces https://github.com/thesofproject/sof/pull/2152 and fixes https://github.com/thesofproject/linux/issues/1561

@plbossart this should fix the issue above but there could be other errors that might need to be handled.

**Comment by juimonen** [comment]
@ranj063 ok nice! and probably the correct fix. I will not approve this yet though if you want to do some qa test for side effects...

**Comment by ranj063** [comment]
> @ranj063 ok nice! and probably the correct fix. I will not approve this yet though if you want to do some qa test for side effects...

@juimonen there will no side effects of adding virtual widgets but what I want to know is if BYT_CR with RT5640 needs any addition virtual widgets. I dont have a device to check this. @plbossart can you please confirm?

File: tools/topology/sof-byt-codec.m4 [hunk 1]
```diff
@@ -98,5 +98,9 @@ VIRTUAL_WIDGET(ssp2 Rx, out_drv, 1)
...
 VIRTUAL_WIDGET(ssp0 Tx, out_drv, 3)
 VIRTUAL_WIDGET(ssp0 Rx, out_drv, 4)
-VIRTUAL_DAPM_ROUTE_IN(modem_in, SSP, 0, IN, 0)
-VIRTUAL_DAPM_ROUTE_OUT(modem_out, SSP, 0, OUT, 1)
+VIRTUAL_WIDGET(modem_out, out_drv, 5)
+VIRTUAL_WIDGET(modem_in, input, 6)
+VIRTUAL_WIDGET(AIF2 Playback, out_drv, 7)
+VIRTUAL_WIDGET(AIF2 Capture, input, 8)
+VIRTUAL_WIDGET(AIF1 Playback, out_drv, 9)
+VIRTUAL_WIDGET(AIF1 Capture, input, 10)
```
Comment (plbossart) [inline tools/topology/sof-byt-codec.m4]: no, those AIF thingies are RT5640 widgets, we need to leave them also.

Also why is all this out_drv? isn't there an in_drv?

and last should we move all this to the byt/cht.m4 files?

**Comment by plbossart** [comment]
> > @ranj063 ok nice! and probably the correct fix. I will not approve this yet though if you want to do some qa test for side effects...
> 
> @juimonen there will be no side effects of adding virtual widgets but what I want to know is if BYT_CR with RT5640 needs any additional virtual widgets. I dont have a device to check this. @plbossart can you please confirm?

no, all I need is this on BYT-CR w/ RT5640.

VIRTUAL_WIDGET(ssp2 Rx, out_drv, 1)
VIRTUAL_WIDGET(ssp2 Tx, out_drv, 2)
VIRTUAL_WIDGET(ssp0 Tx, out_drv, 3)
VIRTUAL_WIDGET(ssp0 Rx, out_drv, 4)
VIRTUAL_WIDGET(modem_out, out_drv, 5)
VIRTUAL_WIDGET(modem_in, out_drv, 6)

**Comment by ranj063** [comment]
@plbossart @juimonen updated now.

File: tools/topology/platform/intel/byt.m4 [hunk 2]
```diff
@@ -34,3 +34,9 @@ VIRTUAL_WIDGET(codec_out0, output, 1)
 VIRTUAL_WIDGET(codec_out1, output, 2)
 VIRTUAL_WIDGET(codec_in0, input, 3)
 VIRTUAL_WIDGET(codec_in1, input, 4)
+VIRTUAL_WIDGET(ssp2 Rx, out_drv, 5)
+VIRTUAL_WIDGET(ssp2 Tx, out_drv, 6)
+VIRTUAL_WIDGET(ssp0 Tx, out_drv, 7)
+VIRTUAL_WIDGET(ssp0 Rx, out_drv, 8)
+VIRTUAL_WIDGET(modem_out, out_drv, 9)
+VIRTUAL_WIDGET(modem_in, input, 10)
```
Comment (plbossart) [inline tools/topology/platform/intel/byt.m4]: this works, but shouldn't we clean the out_drv and input/output to be consistent?

Comment (ranj063) [reply tools/topology/platform/intel/byt.m4]: @plbossart is it better now?

Comment (plbossart) [reply tools/topology/platform/intel/byt.m4]: Dunno. what is the opposite of out_drv? plain vanilla input?

Comment (ranj063) [reply tools/topology/platform/intel/byt.m4]: OK fixed now

**Comment by plbossart** [comment]
validation shows issue #1561 is fixed so we can merge this PR and close PR #2152

**Comment by lgirdwood** [comment]
@zrombel build not found in CI ?

**Comment by zrombel** [comment]
QB issue. I've scheduled it for rebuild.

**Comment by lgirdwood** [comment]
CI usual suspects.

**Pull request merged**